src/mozanalysis/frequentist_stats/linear_models.py

```diff
@@ -0,0 +1,412 @@
+from statsmodels.regression.linear_model import RegressionResults
+from statsmodels.stats.weightstats import DescrStatsW
+
+from mozanalysis.utils import filter_outliers
+
+
+def stringify_alpha(alpha: float) -> tuple[str, str]:
+    """Converts a floating point alpha-level to the string
+    labels of the endpoint of a confidence interval.
+    E.g., 0.05 -> '0.025', '0.975'"""
+    if alpha <= 0 or alpha >= 1:
+        raise ValueError("alpha must be in (0,1)")
+    return f"{alpha/2:0.3f}", f"{1-alpha/2:0.3f}"
+
+
+def summarize_one_branch(branch_data: pd.Series, alphas: list[float]) -> pd.Series:
+    """Inferences (point estimate and confidence intervals) for
+    the mean of a single branch's data. Constructs confidence
+    intervals from central limit theory (uses the t-distribution)
+
+    Parameters:
+    - branch_data (pd.Series): the vector of observations from a
+    single branch.
+    - alphas (list[float]): the desired confidence levels
+
+    Returns:
+    - result (pd.Series): the resulting inferences on the mean. Has
+    the following elements in its index (assuming `alphas=[0.01,0.05]`):
+      - 'mean': the point estimate of the mean
+      - '0.5': also the point estimate, included for backwards compatibility
+    with prior bootstrap implementations
+      - '0.005': the lower bound of the 99% confidence interval
+      - '0.025': the lower bound of the 95% confidence interval
+      - '0.975': the upper bound of the 95% confidence interval
+      - '0.995': the upper bound of the 99% confidence interval
+    """
+    str_quantiles = ["0.5"]
+    for alpha in alphas:
+        str_quantiles.extend(stringify_alpha(alpha))
+    res = pd.Series(index=sorted(str_quantiles) + ["mean"], dtype="float")
+    dsw = DescrStatsW(branch_data)
+    mean = dsw.mean
+    res["0.5"] = mean  # backwards compatibility
+    res["mean"] = mean
+    for alpha in alphas:
+        low, high = dsw.tconfint_mean(alpha)
+        low_str, high_str = stringify_alpha(alpha)
+        res[low_str] = low
+        res[high_str] = high
+    return res
+
+
+def summarize_univariate(
+    data: pd.Series, branches: pd.Series, branch_list: list[str], alphas: list[float]
+) -> dict[str, pd.Series]:
+    """Univariate inferences (point estimates and confidence intervals) for the
+    mean of each branch's data.
+
+    Parameters:
+    - data (pd.Series): the vector of observations
+    - branches (pd.Series): the vector of branch names. Expected to be the same length
+    as `data`
+    - branch_list (list[str]): the list of branches to perform inferences on
+    - alphas (list[float]): the desired confidence levels
+
+    Returns:
+    - out (dict[str, pd.Series]): a dictionary keyed by branch label with a pandas
+    series as value. The format of the series is described above in
+    `summarize_one_branch`
+    """
+    return {
+        b: summarize_one_branch(data.loc[(branches == b).values], alphas)
+        for b in branch_list
```

Comment:
In most cases with `branch_list == branches.unique()`?  If so, maybe make it an optional parameter that defaults to `branches.unique()`

Reply:
I added a function to do this.